builder/ecs/image_config.go

```diff
@@ -178,7 +185,24 @@ func (c *AlicloudImageConfig) Prepare(ctx *interpolate.Context) []error {
 	if reg.FindString(c.AlicloudImageName) != "" {
 		errs = append(errs, fmt.Errorf("image_name can't include spaces"))
 	}
-
+	if c.AlicloudTargetImageFamily != "" {
+		if strings.HasPrefix(c.AlicloudTargetImageFamily, "http://") ||
+			strings.HasPrefix(c.AlicloudTargetImageFamily, "https://") ||
+			strings.HasPrefix(c.AlicloudTargetImageFamily, "acs:") ||
+			strings.HasPrefix(c.AlicloudTargetImageFamily, "aliyun") {
+			errs = append(errs, fmt.Errorf("target_image_family can't start with 'aliyun', 'acs:', 'http://' or 'https://'"))
+		} else {
+			imageFamilyReg := regexp.MustCompile(`^[a-zA-Z\p{L}][a-zA-Z\p{L}_0-9\-\.\:]{1,127}$`)
```

Comment:
I don't know all my unicode character classes by heart, but doesn't L also match the `a-zA-Z` ranges? We can probably simplify that expression if that's the case
```suggestion
			imageFamilyReg := regexp.MustCompile(`^\p{L}[\p{L}_0-9\-\.\:]{1,127}$`)
```
I believe also that both `.` and `:` don't need escaping in the context of a `[]`? Not sure though, that probably depends on the regex engine.

Reply:
After testing, I find that `p{L}` can match `a-zA-Z` but can't match `.` or digit, so I simplify it as your suggestion. It's pleasure for me to learn more about the regex rule.